.github/workflows/precommit.yml

```
@@ -0,0 +1,47 @@
+          echo "go-build=$(go env GOCACHE)" >> "$GITHUB_OUTPUT"
+
+      - name: Go Build Cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-build }}
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}-${{ matrix.os }}-${{ matrix.arch }}
+
+      - name: Run pre-commit hooks
+        env:
+          GOPROXY: direct
+          GOOS: ${{ matrix.os }}
```

actionlint (1.7.4)

property "os" is not defined in object type {}

(expression)